src/figure_settings.py

```diff
@@ -130,13 +129,9 @@ def on_select(self, model, _pos, _n_items):
                 figure_settings.set_use_custom_style(True)
 
     def set_axes_entries(self):
-        used_axes = [False, False, False, False]
-        for item in self.get_application().get_data():
-            for i in item.get_xposition() * 2, 1 + item.get_yposition() * 2:
-                used_axes[i] = True
-        if not any(used_axes):
-            used_axes = [True, True, False, False]
-        for (direction, visible) in zip(_DIRECTIONS, used_axes):
+        canvas = self.get_application().get_window().get_canvas()
+        visible_axes = canvas.visible_axes
```

Comment:
The issue I have with this it that it ties the figure settings to the canvas, when the application is structured the other way around. I'd keep the implementation here

Reply:
The reason for doing it this way was mainly because these theoretically should give the same results, and the old implementation thus kinda is a repetition of the logic in the canvas.

But I am fine with keeping the earlier implementation still. Or perhaps reimplement the logic as a property of the Data would make more sense. That would make it a bit more agnostic to the specific canvas or the application itself.